src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -774,6 +799,34 @@ namespace winrt::TerminalApp::implementation
         focusedTab->ClosePane();
     }
 
+    // Method Description:
+    // - Close the terminal app with keys. If there are more
+    // - than one tab opened, show a warning dialog.
+    void TerminalPage::_CloseWindow()
+    {
+        if (_tabs.size() > 1)
+        {
+            _ShowCloseWarningDialog();
+        }
+        else
+        {
+            _CloseAllTabs();
+        }
+    }
+
+    // Method Description:
+    // - Close all the tabs opened and this will finally terminate
+    // - the terminal
```

Comment:
I'd maybe add that the terminal will terminate on it's own when the last tab is removed, because this method isn't really the one doing the "terminate", it's just doing the "remove all tabs" bit.

Reply:
changed